src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -210,14 +220,32 @@ private void TrackOpenedDocument(uint docCookie, string moniker)
 
         private void Registration_WorkspaceChanged(object sender, EventArgs e)
         {
+            // We may or may not be getting this notification from the foreground thread if another workspace
+            // is raising events on a background. Let's send it back to the UI thread since we can't talk
+            // to the RDT in the background thread. Since this is all asynchronous a bit more asynchrony is fine.
+            if (!_foregroundThreadAffinitization.IsForeground())
+            {
+                ScheduleTask(() => Registration_WorkspaceChanged(sender, e));
```

Comment:
I don't recall, I'd have to look again. This is mostly to protect against other workspaces doing something asynchronously. My change for VisualStudioWorkspace in the other PR does also ensure MiscellaneousFilesWorkspace will get the same behavior since I'm defining it as a host workspace service.